packages/flutter_tools/lib/src/isolated/native_assets/dart_hook_result.dart

```diff
@@ -0,0 +1,114 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:native_assets_cli/code_assets_builder.dart';
+import 'package:native_assets_cli/data_assets_builder.dart';
+import '../../asset.dart' show FlutterHookResult, HookAsset;
+import 'native_assets.dart' show FlutterCodeAsset;
+
+/// The assets produced by a Dart hook run and the dependencies of those assets.
+///
+/// If any of the dependencies change, then the Dart build should be performed
+/// again.
+final class DartHooksResult {
+  const DartHooksResult({
+    required this.buildStart,
+    required this.buildEnd,
+    required this.codeAssets,
+    required this.dataAssets,
+    required this.dependencies,
+  });
+
+  DartHooksResult.empty()
+    : buildStart = DateTime.now(),
+      buildEnd = DateTime.now(),
+      codeAssets = const <FlutterCodeAsset>[],
+      dataAssets = const <DataAsset>[],
+      dependencies = const <Uri>[];
+
+  factory DartHooksResult.fromJson(Map<String, Object?> json) {
```

Comment:
Can we rework this to use pattern matching to deconstruct the JSON response? It should remove the need for a lot of these casts.